### DISCUSSION TITLE
Build within Dockerfile

### DISCUSSION THREAD

**Discussion by lcmgh** [paritytech/cachepot#164] [General]
Hi!

Could I utilize Cachepot to build within a Dockerfile with help of a buildx mounted directory?

Or should I rather rely on something like https://github.com/LukeMathWalker/cargo-chef?

Thanks for your input

**Comment by drahnr** [comment]
I think this needs some more detail on the exact use case to spell out a recommendation on your specific use case.

**Reply by lcmgh** [reply]
Usecase: Improve build time of Docker based Rust builds.

So what I understand:

`cargo chef`:
* caches compiled dependencies
* need to rebuild all dependencies if a single dependency changes
* Caching is limited to Docker builds and scoped to a single project

`cachepot`
* Caches single compiled binaries
* Caching can be utilised across several projects
* Assumption: Works also within Docker builds by using a mount `buildx` flag: `RUN --mount=type=cache,target=$CACHEPOT_DIR cargo build`

I have not verified the latter due to general buildx problems in my environment. I expect that I could enhance build time by utilising both tools.

**Reply by drahnr** [reply]
So you want to run cachepot from within your docker container by mounting both the cache dir of cachepot and the cachepot binary itself? If so, yes, that would work.

**Reply by lcmgh** [reply]
Cool. I would install the cachepot binary within the Dockerfile and then use the cache dir of cachepot as buildx mounted directory.
